### ISSUE TITLE
Support for Alpine's apk tool

### ISSUE THREAD

**Issue by timbru31** [hadolint/hadolint#132]
- [ ] This is a bug report
- [x] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

Personally I prefer to stick to the much smaller alpine tags of any Docker image that supports it. The underlying [Alpine distro](https://alpinelinux.org/) uses it's own package manager, called [apk](https://wiki.alpinelinux.org/wiki/Alpine_Linux_package_management) which has the same features like apt-get, but is currently not linted.

### Actual behavior

It would be a good addition if hadolint was able to lint apk operations, too.  
Possible rules:
* Don't upgrade the system (via `apk upgrade` or `apk update && apk upgrade`)
* Use version pinning like apt-get (via `apk add asterisk=1.6.0.21-r0`)
* Delete cache (needs further investigation)

### Steps to reproduce the behavior

Lint Dockerfile with flaws in it, see no issues reported.

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter UNKNOWN
```

This is weird. Just installed hadolint via Homebrew and got
```bash
==> Pouring hadolint-1.2.3.high_sierra.bottle.tar.gz
```

**Dockerfile**

```Dockerfile
FROM openjdk:8-jre-alpine

RUN apk update && apk upgrade && apk add \
  nodejs \
  nodejs-npm\
  curl
```

**Additional environment details (OS, stack version, etc.)**

* macOS High Sierra 10.13.2 (17C88)
* Homebrew 1.4.0 (Homebrew/homebrew-core (git revision dbce; last commit 2017-12-11))

**Comment by zemanlx** [comment]
@timbru31 I have started working on it yesterday :smile: 

In your example, you use `update` and `upgrade` that I want to ban like we are doing it for `apt-get` in [DL3005](https://github.com/hadolint/hadolint/wiki/DL3005) and suggest use `--no-cache`. 

What is your use case for `apk update && apk upgrade`?

**Comment by harshit22394** [comment]
Hi @zemanlx ....i am facing the same --version issue

⇒  hadolint --version
Haskell Dockerfile Linter UNKNOWN

Can you let us know by what time we can have the issue fixed. Since the hadolint linter on sublime is not working because of this issue.

**Comment by zemanlx** [comment]
@harshit22394 Can you please raise a separate issue for that?

**Comment by harshit22394** [comment]
sure

**Comment by timbru31** [comment]
> What is your use case for apk update && apk upgrade?

Only use case I can imagine is updating the underlying OS to e.g. fix a CVE. Nevertheless the same as for apt-get applies. (maintainer should update the image)

>  I have started working on it yesterday 😄
Great news, I already saw the PR 🎉
